lib/segment/src/types.rs

```diff
@@ -1132,6 +1135,27 @@ impl Payload {
     pub fn contains_key(&self, key: &str) -> bool {
         self.0.contains_key(key)
     }
+
+    pub fn size_bytes(&self) -> usize {
+        let mut size = 0;
+        for (k, v) in &self.0 {
+            size += k.len();
+            size += value_size(v);
+        }
+
+        size
+    }
+}
+
+fn value_size(value: &Value) -> usize {
+    match value {
+        Value::Null => 0,
+        Value::Bool(_) => 1,
+        Value::Number(_) => 8,
```

Comment:
Though it's effectively the same, I do prefer this:

```suggestion
        Value::Null => std::mem::size_of::<()>(),
        Value::Bool(_) => std::mem::size_of::<bool>(),
        Value::Number(_) => std::mem::size_of::<f64>(),
```